Tasks/AzureCLIV1/azureclitask.ts

```diff
@@ -98,11 +98,24 @@ export class azureclitask {
 
     private static loginAzureRM(connectedService: string): void {
         var servicePrincipalId: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalid", false);
-        var servicePrincipalKey: string = tl.getEndpointAuthorizationParameter(connectedService, "serviceprincipalkey", false);
+        let authType: string = tl.getEndpointAuthorizationParameter(connectedService, 'authenticationType', true);
+        let cliPassword: string = null;
+        if(authType == "spnCertificate") {
+            tl.debug('certificate based endpoint');
+            let certificateContent: string = tl.getEndpointAuthorizationParameter(connectedService, "servicePrincipalCertificate", false);
+            cliPassword = path.join(tl.getVariable('Agent.TempDirectory') || tl.getVariable('system.DefaultWorkingDirectory'), 'spnCert.pem');
```

Comment:
We need to delete the file after login

Reply:
+1

Reply:
done. thanks

Reply:
Agent with > 2.115.1 saves the file in Agent.tempDirectory which removes the file after the task completion.
For others, I have added the process of removing the file manually.